src/main/java/org/ow2/proactive/connector/iaas/cloud/provider/azure/AzureProvider.java

```diff
@@ -200,13 +220,14 @@
         Optional<NetworkSecurityGroup> optionalNetworkSecurityGroup = options.map(Options::getSecurityGroupNames)
                                                                              .map(secGrpNames -> secGrpNames.get(0))
                                                                              .map(secGrpName -> azureProviderUtils.searchNetworkSecurityGroupByName(azureService,
+                                                                                                                                                    resourceGroup.name(),
                                                                                                                                                     secGrpName)
                                                                                                                   .get());
 
         // Get existing public IP address if specified
-        Optional<PublicIpAddress> optionalPublicIpAddress = options.map(Options::getPublicIpAddress)
-                                                                   .map(publicIpAddress -> azureProviderUtils.searchPublicIpAddressByIp(azureService,
-                                                                                                                                        publicIpAddress)
+        Optional<PublicIPAddress> optionalPublicIPAddress = options.map(Options::getPublicIpAddress)
+                                                                   .map(PublicIPAddress -> azureProviderUtils.searchPublicIpAddressByIp(azureService,
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Rename this local variable to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00117)